src/main/java/io/kestra/plugin/nats/Consume.java

```diff
@@ -0,0 +1,141 @@
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Consume messages from a NATS subject on a JetStream-enabled NATS server",
+    description = """
+        Please note that the server you run it against must have JetStream enabled for it to work.
+        It should also have a stream configured to match the given subject."""
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Consume messages from any topic subject matching the kestra.> wildcard, using user password authentication",
+            code = {
+                "url: nats://localhost:4222",
+                "username: nats_user",
+                "password: nats_passwd",
+                "subject: kestra.>",
+                "durableId: someDurableId",
+                "pollDuration: PT5S"
+            }
+        ),
+    }
+)
+public class Consume extends NatsConnection implements RunnableTask<Consume.Output>, ConsumeInterface {
+    private String subject;
+    private String durableId;
+    private String since;
+    @Builder.Default
+    private Duration pollDuration = Duration.ofSeconds(2);
+    @Builder.Default
+    private Integer batchSize = 10;
+    private Integer maxRecords;
+    private Duration maxDuration;
+    @Builder.Default
+    private DeliverPolicy deliverPolicy = DeliverPolicy.All;
+
+    public Output run(RunContext runContext) throws Exception {
+        Connection connection = connect(runContext);
+        JetStreamSubscription subscription = connection.jetStream(JetStreamOptions.DEFAULT_JS_OPTIONS).subscribe(
+            runContext.render(subject),
+            PullSubscribeOptions.builder()
+                .configuration(ConsumerConfiguration.builder()
+                    .ackPolicy(AckPolicy.Explicit)
+                    .deliverPolicy(deliverPolicy)
+                    .startTime(Optional.ofNullable(since).map(throwFunction(ignored -> ZonedDateTime.parse(runContext.render(since)))).orElse(null))
+                    .build())
+                .durable(runContext.render(durableId)).build()
+        );
+
+        Instant pollStart = Instant.now();
+        List<Message> messages;
+        AtomicInteger total = new AtomicInteger();
+        File outputFile = runContext.tempFile(".ion").toFile();
+        try (OutputStream output = new BufferedOutputStream(new FileOutputStream(outputFile))) {
+            AtomicReference<Integer> maxMessagesRemainingRef = new AtomicReference<>();
+            do {
+                Integer maxMessagesRemaining = Optional.ofNullable(maxRecords).map(ignored -> maxRecords - total.get()).orElse(null);
```

Comment:
```suggestion
                Integer maxMessagesRemaining = Optional.ofNullable(maxRecords).map(max -> max - total.get()).orElse(null);
```

As a best practice, always use the value from the `map()` instead of the variable. This improves memory usage as the lambda will be non-capturing and avoid future bugs when refactoring.

Please do this also for the other usage just bellow.